sql/2021/caching/valid_if_modified_since_returns_304.sql

```diff
@@ -0,0 +1,26 @@
+#standardSQL
+# Whether making a If-Modified-Since request returns a 304 if the content have not changed (as seen from Last-Modified)
+SELECT
+  client,
+  COUNT(0) AS total_requests,
+  COUNTIF(status = 304) AS total_304,
+  COUNTIF(NOT uses_etag AND uses_last_modified AND uses_if_modified AND no_change) AS total_expected_304,
+  COUNTIF(NOT uses_etag AND uses_last_modified AND uses_if_modified AND no_change AND status = 304) AS total_actual_304,
+  COUNTIF(status = 304) / COUNT(0) AS pct_304,
+  COUNTIF(NOT uses_etag AND uses_last_modified AND uses_if_modified AND no_change) / COUNTIF(status = 304) AS pct_expected_304,
+  COUNTIF(NOT uses_etag AND uses_last_modified AND uses_if_modified AND no_change AND status = 304) / COUNTIF(NOT uses_etag AND uses_last_modified AND uses_if_modified AND no_change) AS pct_actual_304
```

Comment:
Using the same complex expression multiple times (sometimes with a small tweak so not the same). Would it be better to refactor to another subquery with a named column to make this clearer and more obvious?

Reply:
This is another case of copying a 2020 query for consistency